contracts/cw-staking/src/traits/cw_staking_adapter.rs

```diff
@@ -30,6 +32,12 @@ pub trait CwStakingAdapter: Identify {
             .map_err(Into::into)
     }
 
+    /// Build the LP token from the provided asset entry
+    fn provider_lp_token(&self, staking_asset: &AssetEntry) -> StdResult<LpToken> {
```

Comment:
I thought lp tokens already contained the dex they refer to? how else are we going to reference lp tokens that have the same composition but different dexes? 

See our doc on naming -> https://github.com/Abstract-OS/contracts/blob/main/development/NAMING.md

Reply:
The lp token that is passed into the `info` query is:

`terra2>astro,terra2>luna` which is NOT an LP token because it doesn't have the dex that it refers to. The dex is passed into another argument:

```rust
    Info {
        provider: ProviderName,
        staking_token: AssetEntry,
    },
```

Reply:
imo it should be, the LP token is an implicit AssetEntry (it can be resolved as one). 
The provider and dex of the lp token can be two different things (remember when Astroport was giving emissions for Terraswap LP tokens?)  In that scenario the provider is Astroport and the staking token would be something like `terraswap/terra>ust,terra>luna` resulting in a staking contract key of `{ provider: "astroport", contract: "staking/terraswap/terra>ust,terra>luna` }

Reply:
`cw-staking` should not be exclusive to dexes, any staking provider should be supported, hence the explicit separation of the token and the staking provider.

Reply:
Alright then, so the bug present is actually the `staking_contract_name` not the `lp_token`.